kiwi/boot/functions.sh

```diff
@@ -5524,19 +5524,26 @@ function waitForStorageDevice {
     # function to check access on a storage device
     # which could be a whole disk or a partition.
     # the function will wait until the size of the
-    # storage device could be obtained or the check
-    # counter equals 4
+    # storage device could be obtained or the timeout
+    # is reached. Default timeout is 60 seconds, however
+    # it can be set to different value by setting the
+    # DEVICE_TIMEOUT variable in the kernel command
+    # line.
     # ----
     local IFS=$IFS_ORIG
     local device=$1
     local check=0
+    local limit=30
+    if [ -n ${DEVICE_TIMEOUT##*[!0-9]*} ]; then
```

Comment:
Yes you are right, there is a mistake here :disappointed:  I missed to add double quotes so it is evaluated as string. The following works:

```bash
if [ -n "${DEVICE_TIMEOUT##*[!0-9]*}" ]; then
    echo $DEVICE_TIMEOUT
fi
```
I have not used double brackets and regexp since I could not easily figure out it is not Bash specific syntax and I was not sure this script is running under a bash shell. I am totally fine with regexp, we can change it if is it going to be more consistent with the rest of the code.

About not being under 2 seconds... I think I would not care, it would result in 0 seconds timeout instead of 1 second (something that we can't achieve unless we modify our waiting steps to `sleep 1`). Moreover someone that uses DEVICE_TIMEOUT=1 I think it is going to be aware enough to increase it if something fails due to not waiting enough. Or, on the other hand, we could ceil the number with something like:

```bash
limit=$(((DEVICE_TIMEOUT + 1) / 2))
```